apps/couchbase.go

```diff
@@ -0,0 +1,179 @@
+func (lr LoggingReceiverCouchbase) Components(tag string) []fluentbit.Component {
+	if len(lr.IncludePaths) == 0 {
+		lr.IncludePaths = []string{
+			"/opt/couchbase/var/lib/couchbase/logs/couchdb.log",
+			"/opt/couchbase/var/lib/couchbase/logs/info.log",
+			"/opt/couchbase/var/lib/couchbase/logs/debug.log",
+			"/opt/couchbase/var/lib/couchbase/logs/error.log",
+			"/opt/couchbase/var/lib/couchbase/logs/babysitter.log",
+		}
+	}
+	lr.MultilineRules = []confgenerator.MultilineRule{
+		{
+			StateName: "start_state",
+			NextState: "cont",
+			Regex:     `^\[([^\s+:]*):`,
+		},
+		{
+			StateName: "cont",
+			NextState: "cont",
+			Regex:     `^(?!\[([^\s+:]*):).*$`,
+		},
+	}
+	components := lr.LoggingReceiverFilesMixin.Components(tag)
+	components = append(components, confgenerator.LoggingProcessorParseRegex{
+		Regex: `^\[(?<type>[^:]*):(?<severity>[^,]*),(?<timestamp>\d+-\d+-\d+T\d+:\d+:\d+.\d+Z),(?<node_name>[^:]*):(?<module_name>[^\<]+)(?<source>[^\]]+)\](?<message>.*)$`,
+		ParserShared: confgenerator.ParserShared{
+			TimeKey:    "timestamp",
+			TimeFormat: "%Y-%m-%dT%H:%M:%S.%L",
+		},
+	}.Components(tag, "couchbase_default")...)
+
+	components = append(components,
+		fluentbit.TranslationComponents(tag, "severity", "logging.googleapis.com/severity", false,
+			[]struct{ SrcVal, DestVal string }{
+				{
+					"debug", "DEBUG",
+				},
+				{
+					"info", "INFO",
+				},
+				{
+					"warn", "WARNING",
+				},
+				{
+					"error", "ERROR",
+				},
+			},
+		)...,
+	)
+	return components
+}
+
+// LoggingProcessorCouchbaseHTTPAccess is a struct that will generate the fluentbit components for the http access logs
+type LoggingProcessorCouchbaseHTTPAccess struct {
+	confgenerator.ConfigComponent           `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+// Type returns the string for the couchbase http access logs
+func (lp LoggingProcessorCouchbaseHTTPAccess) Type() string {
+	return "couchbase_http_access"
+}
+
+// Components returns the fluentbit components for the http access logs of couchbase
+func (lp LoggingProcessorCouchbaseHTTPAccess) Components(tag string) []fluentbit.Component {
+	if len(lp.IncludePaths) == 0 {
+		lp.IncludePaths = []string{
+			"/opt/couchbase/var/lib/couchbase/logs/http_access.log",
+			"/opt/couchbase/var/lib/couchbase/logs/http_access_internal.log",
+		}
+	}
+	c := lp.LoggingReceiverFilesMixin.Components(tag)
+	c = append(c,
+		confgenerator.LoggingProcessorParseRegex{
+			Regex: `^(?<client_ip>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<timestamp>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^ ]*) +\S*)?" (?<status_code>[^ ]*) (?<response_size>[^ ]*) - (?<message>.*)$`,
+			ParserShared: confgenerator.ParserShared{
+				TimeKey:    "timestamp",
+				TimeFormat: `%d/%b/%Y:%H:%M:%S %z`,
+				Types: map[string]string{
+					"size": "integer",
+					"code": "integer",
+				},
+			},
+		}.Components(tag, "couchbase_http_access")...,
```

Comment:
We will need to add the Instrumentation source label here too no? See previous comment.